sched/sched/sched_lock.c

```diff
@@ -46,6 +46,30 @@
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name:  sched_lock_wo_note
+ *
+ * Description:
+ *   This function disables context switching.
+ *   It does not perform instrumentation logic.
+ *
+ ****************************************************************************/
+
+bool sched_lock_wo_note(void)
+{
+  FAR struct tcb_s *tcb;
+
+  if (OSINIT_TASK_READY() && !up_interrupt_context())
+    {
+      tcb = this_task();
+      tcb->lockcount++;
+
+      return tcb->lockcount == 1 ? true : false;
```

Comment:
race condition, lock count should use atomic

Reply:
lockcount is only modified and read by the current task, and there are no race condition here